hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/token/delegation/AbstractDelegationTokenSecretManager.java

```diff
@@ -102,26 +102,26 @@ private String formatTokenId(TokenIdent id) {
    * Protected by this object lock.
    */
   protected int delegationTokenSequenceNumber = 0;
-  
+
   /**
    * Access to allKeys is protected by this object lock
    */
-  protected final Map<Integer, DelegationKey> allKeys 
+  protected final Map<Integer, DelegationKey> allKeys
       = new ConcurrentHashMap<>();
-  
+
   /**
    * Access to currentId is protected by this object lock.
    */
   protected int currentId = 0;
   /**
    * Access to currentKey is protected by this object lock
    */
-  private DelegationKey currentKey;
-  
-  private long keyUpdateInterval;
-  private long tokenMaxLifetime;
-  private long tokenRemoverScanInterval;
-  private long tokenRenewInterval;
+  private volatile DelegationKey currentKey;
```

Comment:
I have set it to volatile to make sure currentKey changes are visible across threads